File: content/blocks/java/balanced-brackets.java
static boolean isBalanced(String code) {
  Deque<Character> stack = new ArrayDeque<>();
  for (char c : code.toCharArray()) {
    switch (c) {
      case '(', '[', '{' -> stack.push(c);
      case ')', ']', '}' -> {
        char open = c == ')' ? '(' : c == ']' ? '[' : '{';
        if (stack.isEmpty() || stack.pop() != open) {
          return false;
        }
      }
      default -> {}
    }
  }
  return stack.isEmpty();
}

File: content/blocks/go/balanced-brackets.go
func balanced(code string) bool {
	pairs := map[rune]rune{')': '(', ']': '[', '}': '{'}
	var stack []rune
	for _, r := range code {
		switch r {
		case '(', '[', '{':
			stack = append(stack, r)
		case ')', ']', '}':
			if len(stack) == 0 || stack[len(stack)-1] != pairs[r] {
				return false
			}
			stack = stack[:len(stack)-1]
		}
	}
	return len(stack) == 0
}
